benchmark/benchmark.go

```diff
@@ -30,3 +54,101 @@ func GetSubject() (string, error) {
 	}
 	return subject, nil
 }
+
+// Wrapper reports the benchmarking metrics of targetFunc to testing.B.
+func Wrapper(b *testing.B, targetFunc func(), cleanupFunc func()) {
+	metricsSum := Metrics{}
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		b.StartTimer()
+		metrics, err := measureMetrics(targetFunc)
+		if err != nil {
+			b.Fatal("Error measuring metrics:", err)
+			return
+		}
+		b.StopTimer()
+		metricsSum = addMetrics(metricsSum, metrics)
+		cleanupFunc()
+	}
+	b.ReportMetric(metricsSum.PeakCPUUsage/float64(b.N), "%cpu_peak/op")
+	b.ReportMetric(metricsSum.AverageCPUUsage/float64(b.N), "%cpu_avg/op")
+	b.ReportMetric(metricsSum.TotalCPUTime.Seconds()/float64(b.N), "cpu_seconds/op")
+	b.ReportMetric(float64(metricsSum.DiskUsageDelta/int64(b.N)), "disk_bytes/op")
+}
+
+func measureMetrics(f func()) (Metrics, error) { //nolint:unparam // make it extensible for future error handling
+	done := make(chan struct{})
+	var cpuUsage []float64
+	var startTime time.Time
+	var diskUsageBefore, diskUsageAfter uint64
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+
+	go func() {
+		f()
+		done <- struct{}{}
+	}()
+
+	go func() {
+		before, err := disk.Usage("/")
+		if err != nil {
+			fmt.Printf("Error getting disk usage before: %v\n", err)
+			wg.Done()
+			return
+		}
+		diskUsageBefore = before.Free
+
+		ticker := time.NewTicker(10 * time.Millisecond)
+		startTime = time.Now()
+
+		for {
+			select {
+			case <-ticker.C:
+				percent, err := cpu.Percent(time.Millisecond*10, false)
+				if err != nil {
+					ticker.Stop()
+					wg.Done()
+					return
+				}
+				cpuUsage = append(cpuUsage, percent[0])
+			case <-done:
+
+				after, err := disk.Usage("/")
+				if err != nil {
+					fmt.Printf("Error getting disk usage after: %v\n", err)
```

Comment:
Instead of fmt.printf(), should we use logrus? that way we can print errors as Warning/Error. which would be easy to find?